From 09a779feae2c14ef309e5057229cce4ed7bbf5df Mon Sep 17 00:00:00 2001
From: johnlu <chengjun.lu@intel.com>
Date: Tue, 26 Apr 2022 00:56:10 +0000
Subject: [PATCH 09/28] Fix issue in sparce_coo_tensor only supporting CUDA
 device.

## Motivation
The at::native::_validate_sparse_coo_tensor_args only supports checking the indices on CUDA device and CPU device. To extend the function to support more device type.

## Solution
Copy  the indices to the CPU to validate the correctness.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/76293
Approved by: https://github.com/mrshenli
---
 aten/src/ATen/native/sparse/SparseTensor.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aten/src/ATen/native/sparse/SparseTensor.cpp b/aten/src/ATen/native/sparse/SparseTensor.cpp
index 90a2c68299..ecc0c2784a 100644
--- a/aten/src/ATen/native/sparse/SparseTensor.cpp
+++ b/aten/src/ATen/native/sparse/SparseTensor.cpp
@@ -295,7 +295,7 @@ void _validate_sparse_coo_tensor_args(
     Tensor max_indices =
         std::get</* values */ 0>(indices.max(/* dim */ 1, /* keepdim */ false));
     Tensor cpu_min_indices, cpu_max_indices;
-    if (indices.is_cuda()) {
+    if (!indices.is_cpu()) {
       cpu_min_indices = min_indices.to(at::DeviceType::CPU);
       cpu_max_indices = max_indices.to(at::DeviceType::CPU);
     } else {
-- 
2.25.1

